api/src/plugins/cors.ts

```diff
@@ -0,0 +1,41 @@
+import { FastifyPluginCallback } from 'fastify';
+
+import fp from 'fastify-plugin';
+import { HOME_LOCATION } from '../utils/env';
+
+// import { FREECODECAMP_NODE_ENV } from '../utils/env';
+
+const allowedOrigins = [
+  'https://www.freecodecamp.dev',
+  'https://www.freecodecamp.org',
+  'https://beta.freecodecamp.dev',
+  'https://beta.freecodecamp.org',
+  'https://chinese.freecodecamp.dev',
+  'https://chinese.freecodecamp.org'
+];
+
+const cors: FastifyPluginCallback = (fastify, _options, done) => {
+  fastify.addHook('onRequest', async (req, reply) => {
+    const origin = req.headers.origin;
+    if (origin && allowedOrigins.includes(origin)) {
+      void reply.header('Access-Control-Allow-Origin', origin);
+    } else {
+      // TODO: Discuss if this is the correct approach. Standard practice is to
+      // reflect one of a list of allowed origins and handle development
+      // separately. If we switch to that approach we can replace use
+      // @fastify/cors instead.
```

Comment:
Should CORS not be mounted conditionally, such that it is disabled in development entirely?

Reply:
There are a few things we can do, but we need to provide something for that header (either '*' or the actual origin). If we don't, any browser will throw CORS errors. At least, that's my understanding.

I think

> if development, then add Access-Control-Allow-Origin: origin

should be fine, but it's a departure from our current approach.